src/FlagBag.php

```diff
@@ -81,6 +83,14 @@ public static function fromAll(string $enumType): FlagBag
 
     /**
      * @param class-string<T>|T $enumOrType
+     *
+     * @phpstan-return FlagBag<T>
+     *
+     * @psalm-template E of \BackedEnum
+     *
+     * @psalm-param E ...$flags
+     *
+     * @psalm-return FlagBag<E>
```

Comment:
Why does it need another template here? Can't it be:

```suggestion
     *
     * @template E of \BackedEnum
     *
     * @param class-string<E>|E $enumOrType
     * @param E[] ...$flags
     *
     * @return FlagBag<E>
```

? The backed enum type is supposed to be the same

Reply:
Please see https://github.com/Elao/PhpEnums/issues/192#issuecomment-1573722202